src/hb-ot-cmap-table.hh

```diff
@@ -272,17 +272,12 @@ struct CmapSubtableLongSegmented
   }
 
   inline bool serialize(hb_serialize_context_t *context,
-                        unsigned int group_count,
-                        Supplier<CmapSubtableLongGroup> &group_supplier)
+                        hb_prealloced_array_t<CmapSubtableLongGroup> &group_data)
   {
     TRACE_SERIALIZE (this);
     if (unlikely(!context->extend_min (*this))) return_trace (false);
-    if (unlikely(!groups.serialize(context, group_count))) return_trace (false);
-    for (unsigned int i = 0; i < group_count; i++)
-    {
-      const CmapSubtableLongGroup &group = group_supplier[i];
-      memcpy(&groups[i], &group, sizeof(group));
-    }
+    if (unlikely(!groups.serialize(context, group_data.len))) return_trace (false);
+    memcpy(&groups[0], &group_data[0], group_data.len * sizeof(CmapSubtableLongGroup));
```

Comment:
There's actually a version of groups.serialize() that takes a Supplier and does the copy. You can make a Supplier here on-demand and pass to that. Like:
```
if (unlikely(!groups.serialize(context, Supplier<...>(group_data.array, group_data.len), group_data.len))) return_trace (false);
```
then you wouldn't need the `memcpy`.

Reply:
Done